main.tf

```diff
@@ -13,6 +13,39 @@ module "label" {
   tags       = "${var.tags}"
 }
 
+resource "aws_s3_bucket" "cache_bucket" {
+  bucket_prefix = "${local.b_name_hyphen}"
+  acl    = "private"
+  force_destroy = true
+  tags = "${module.label.tags}"
+
+  lifecycle_rule {
+    id      = "codebuildcache"
+    enabled = true
+
+    prefix  = "/"
+    tags = "${module.label.tags}"
+
+    expiration {
+      days = "${var.cache_timeout}"
+    }
+  }
+
+}
+
+locals {
+  b_name = "${module.label.id}"
+  b_name_hyphen = "${substr(join("-", split("_", lower(local.b_name))), 0, 20)}"
```

Comment:
Hrm, I think this should be using the `terraform-null-label` module to generate names.

Reply:
What's the `20` character limit based on? (bucket names are 63 characters)

Reply:
Isn't it using terraform-null-label to generate the name?

Reply:
I have shrunk it to 20 characters because it uses `name-prefix` and that appends a long string, which on other builds has caused the name to be longer than 63 chars.
So its playing safe.

Reply:
Aha, I see what you mean. So it looks like you're doing some extra normalization here, which in itself I don't think is a bad idea. I think this is probably out-of-scope, however, for this module. We could consider adding this normalization to https://github.com/cloudposse/terraform-null-label

My concern is that adding it to this module would make the naming convention inconsistent with the rest of the 70+ modules. If we added the contention to the `terraform-null-label` module, then there's a way to start enforcing it throughout the rest of the modules.

@aknysh thoughts?

Reply:
(Also, the caller invoking this module can still noramlize the name the way you do, so no functionality is lost)

Reply:
I'm normalising to make it fit the s3 rules.
But for the `tags` it still uses the unnormalised values.

You could add an `s3` output to the `terraform-null-label`
which normalises for the global s3 bucket naming standard

Reply:
> I have shrunk it to 20 characters because it uses bucket_prefix and that appends a long string, which on other builds has caused the name to be longer than 63 chars.
So its playing safe.

I can certainly see the use-case. 

Could I trouble you make it configurable: `cache_bucket_prefix_enabled`. That way we can keep pretty bucket names for those of us who are particular about those things =)